### PR TITLE
doc: move scheduling to its own section

### PR DESCRIPTION
**Pull request by nashif**
Move scheduling section from under threads and adds some diagrams
showing the different scheduling algorithms we support.

The diagrams were made using draw.io and can be seen here:

https://drive.google.com/file/d/1XiFXC5DlqgjxDg4JSRUnrtfl_LbN45al/view?usp=sharing

### CODE REVIEW

**Review by ioannisg — APPROVED** [review]

**Comment by nashif** [comment]
https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/17522/docs/reference/kernel/scheduling/index.html

**Review by aescolar — COMMENTED** [review]
Thanks for this. A couple of nits:
* It would make reviewing faster if you did this kind of changes (moving something, and doing some minor change to it) in separate commits.
* The figure you link to from the commit description does not seem to be anywhere in the doc itself? (Anyhow, it is a bit funny to link to google drive from a commit message)

**Comment by nashif** [comment]
> It would make reviewing faster if you did this kind of changes (moving something, and doing some minor change to it) in separate commits.

Can be done.

>     * The figure you link to from the commit description does not seem to be anywhere in the doc itself? (Anyhow, it is a bit funny to link to google drive from a commit message)

i am linking to a document with many figures, the one being shown is just the first tab and I have not included it yet, needs some work still.
I see no issue including the google doc in the commit message, although I could also link it in the document as a comment, the idea is to have something that others can change in the future using draw,io.

**Comment by aescolar** [comment]
>  that others can change in the future using draw,io.

Haven't you included the drawio source inside the svg itself?

**Comment by nashif** [comment]
> Haven't you included the drawio source inside the svg itself?

now that you mention it, I think I did while saving, did not realize it would go into the svg and be editable in draw.io later. That is even better.

File: doc/reference/kernel/scheduling/index.rst [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+a cooperative thread, and vice versa, by changing its priority.
+
+The kernel supports a virtually unlimited number of thread priority levels.
+The configuration options :option:`CONFIG_NUM_COOP_PRIORITIES` and
+:option:`CONFIG_NUM_PREEMPT_PRIORITIES` specify the number of priority
+levels for each class of thread, resulting the following usable priority
+ranges:
+
+* cooperative threads: (-:option:`CONFIG_NUM_COOP_PRIORITIES`) to -1
+* preemptive threads: 0 to (:option:`CONFIG_NUM_PREEMPT_PRIORITIES` - 1)
```
Comment (dbkinder) [inline doc/reference/kernel/scheduling/index.rst]: Picture says ``CONFIG_NUM_PREEMPT_PRIORITIES + 1`` but the text here says ``CONFIG_NUM_PREEMPT_PRIORITIES- 1``

**Review by dbkinder — CHANGES_REQUESTED** [review]
One typo in a drawing.

Visually inspecting, it looks like this new doc has identical content as the previous scheduling section, with the addition of the three pictures.  And yes, the .svg files created by draw.io (and saved here in the repo) are editable by draw.io (they include the google "diagram" tags it needs) while also being a renderable .svg files.  So nice.  

We do need to define some drawing standards for colors and so fonts are readable (font choice, size) in the rendered images (particularly for smaller devices), but looks good for now.

**Review by dbkinder — APPROVED** [review]
LGTM

**Pull request merged**